packages/ai-jsx/src/lib/asr/asr.tsx

```diff
@@ -0,0 +1,428 @@
+'use client';
+
+const AUDIO_WORKLET_SRC = `
+class InputProcessor extends AudioWorkletProcessor {
+  constructor() {
+    super();
+  }
+  process(inputs, outputs, parameters) {
+    const input = inputs[0];
+    const output = outputs[0];
+
+    for (let channel = 0; channel < input.length; ++channel) {
+      const inputChannel = input[channel];
+      const outputChannel = output[channel];
+
+      for (let i = 0; i < inputChannel.length; ++i) {
+        outputChannel[i] = inputChannel[i];
+      }
+    }
+
+    // Copy the input data to a new Float32Array
+    const data = new Float32Array(input[0]);
+
+    // Post the data back to the main thread
+    this.port.postMessage(data);
+
+    return true;
+  }
+}
+
+registerProcessor("input-processor", InputProcessor);
+`;
+
+class EventTargetImpl implements EventTarget {
```

Comment:
Can we just use EventEmitter for this? https://nodejs.org/api/events.html

Reply:
note that this code is going to run in-browser